packages/api/cms-api/src/page-tree/entities/page-tree-node-base.entity.ts

```diff
@@ -1,28 +1,26 @@
-import { BaseEntity, Entity, Enum, OptionalProps, PrimaryKey, Property } from "@mikro-orm/core";
+import { BaseEntity, Entity, Enum, Index, ManyToOne, OptionalProps, PrimaryKey, Property } from "@mikro-orm/core";
 import { Field, ID, Int, ObjectType } from "@nestjs/graphql";
 import { v4 as uuid } from "uuid";
 
-import { PageTreeNodeCategory, PageTreeNodeVisibility } from "../types";
+import { PageTreeNodeCategory, PageTreeNodeInterface, PageTreeNodeVisibility } from "../types";
 
 @Entity({ abstract: true })
 @ObjectType("PageTreeNodeBase", { isAbstract: true }) // ObjectType must be defined in base class! (The name "PageTreeNodeBase" is not used (we have no concrete type of PageTreeNodeBase))
-export abstract class PageTreeNodeBase extends BaseEntity<PageTreeNodeBase, "id"> {
+export abstract class PageTreeNodeBase extends BaseEntity {
     static tableName = "PageTreeNode";
     [OptionalProps]?: "createdAt" | "updatedAt" | "hideInMenu";
 
     @PrimaryKey({ columnType: "uuid" })
     @Field(() => ID)
     id: string = uuid();
 
-    @Property({ columnType: "uuid", nullable: true })
+    @Property({ columnType: "uuid", nullable: true, persist: false })
     @Field(() => String, { nullable: true })
     parentId: string | null;
```

Comment:
Do we still need the `parentId`? Wasn't it a substitute for the actual `parent` relation?

Reply:
One benefit of the parentId column is that we don't have to load the parent relation. I could probably change this to a reference, then it should work 🤔

Reply:
This would also cause the parentId to be removed from the GraphQL API. I'd like to do this in a follow-up PR.